### PR TITLE
DWR-303 Add widget to find student by SSID to homepage.

### PR DESCRIPTION
**Pull request by wtritch**
I've also added a new "test" directory to contain re-usable mock classes.  Right now it just contains a mock.data.service.ts file/class.

### CODE REVIEW

File: webapp/src/main/webapp/src/app/student/student.component.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<form [formGroup]="searchForm" (ngSubmit)="performSearch()" class="form-group">
+  <div class="form-group">
+    <div class="flex-children">
+      <div class="flex-child grow">
+        <label for="student-id-input">{{'labels.student.title' | translate}}</label>
+        <input formControlName="ssid" type="text" class="form-control" id="student-id-input" placeholder="{{'labels.student.prompt' | translate}}">
+      </div>
+      <div class="flex-child">
+        <button type="submit" class="btn btn-primary">{{'buttons.search' | translate}}</button>
```
Comment (bgsavage) [inline webapp/src/main/webapp/src/app/student/student.component.html]: Should disable submit if form is not valid?

Comment (esotrope) [inline webapp/src/main/webapp/src/app/student/student.component.html]: Haven't tried out the current impl but yeah i think each form control needs to be filled in sequence and if a previous item is not filled the following items should be disabled

Comment (wtritch) [inline webapp/src/main/webapp/src/app/student/student.component.html]: works for me, I'll disable submit unless there's something in the input

File: webapp/src/main/webapp/src/test/mock.data.service.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+
+import Spy = jasmine.Spy;
```
Comment (bgsavage) [inline webapp/src/main/webapp/src/test/mock.data.service.ts]: This is a great start.  Eventually I think we should just having a TestingModule that other tests imports.

File: webapp/src/main/webapp/src/app/student/student.component.html (hunk 1 above)
Comment (wtritch) [inline webapp/src/main/webapp/src/app/student/student.component.html]: done

**Pull request merged**